### PR TITLE
Fixes Riot Dart Damage Exploit

### LINKED ISSUE #71084: potential foam dart damage exploit
```
/obj/item/ammo_casing/caseless/foam_dart/attackby(obj/item/A, mob/user, params)
	var/obj/projectile/bullet/reusable/foam_dart/FD = loaded_projectile
	if (A.tool_behaviour == TOOL_SCREWDRIVER && !modified)
		modified = TRUE
		FD.modified = TRUE
		FD.damage_type = BRUTE
		to_chat(user, span_notice("You pop the safety cap off [src]."))
		update_appearance()
	else if (istype(A, /obj/item/pen))
		if(modified)
			if(!FD.pen)
				harmful = TRUE
				if(!user.transferItemToLoc(A, FD))
					return
				FD.pen = A
				FD.damage = 5
				FD.nodamage = FALSE
				to_chat(user, span_notice("You insert [A] into [src]."))
			else
				to_chat(user, span_warning("There's already something in [src]."))
		else
			to_chat(user, span_warning("The safety cap prevents you from inserting [A] into [src]."))
	else
		return ..()
```
```
/obj/projectile/bullet/reusable/foam_dart/handle_drop()
	if(dropped)
		return
	var/turf/T = get_turf(src)
	dropped = 1
	var/obj/item/ammo_casing/caseless/foam_dart/newcasing = new ammo_type(T)
	newcasing.modified = modified
	var/obj/projectile/bullet/reusable/foam_dart/newdart = newcasing.loaded_projectile
	newdart.modified = modified
	if(modified)
		newdart.damage = 5
		newdart.nodamage = FALSE
	newdart.damage_type = damage_type
	if(pen)
		newdart.pen = pen
		pen.forceMove(newdart)
		pen = null
	newdart.update_appearance()
```
From what I can tell from the code here, if you tweak a foam dart with a screwdriver but don't insert a pen in it, then fire the foam dart, then recollect it, it will then deal 5 brute damage on future firings, which it's not supposed to do without a pen. This is because handle_drop() incorrectly checks for if(modified) instead of if(pen) during the transfer of some of the dart's qualities.

Other things that might be worth investigating:
* how the setting of the harmful variable on the casing interacts with handle_drop() and with pens that are stuck to your hand
* tk shenanigans to teleport a pen from a foam dart to your hand
* someone on skyrat reported pen'd riot darts dealing cellular damage somehow??? hopefully it's just someone mistaking stamina or oxyloss damage for cellular damage or something

### PR DESCRIPTION
**Pull request by SkeletalElite**
Fixes an issue where riot darts would do damage even if they had no pen in them if recovered after being fired while modified to be able to accept pens
## About The Pull Request
Fixes an issue where riot darts would do damage even if they had no pen in them if recovered after being fired while modified to be able to accept pens
Fixes #71084 
## Why It's Good For The Game
bug bad
## Changelog
:cl:
fix: Riot darts no longer do damage with no pen inserted if recovered after being fired while modified to accept pens. 
/:cl:

### CODE REVIEW

File: code/modules/projectiles/projectile/reusable/foam_dart.dm [hunk 1]
```diff
@@ -21,14 +21,13 @@
 	newcasing.modified = modified
```
Comment (ATH1909) [inline code/modules/projectiles/projectile/reusable/foam_dart.dm:21]:
```suggestion
	newcasing.modified = modified
	newcasing.harmful = harmful
```

File: code/modules/projectiles/projectile/reusable/foam_dart.dm [hunk 2]
```diff
@@ -21,14 +21,13 @@
...
-	if(modified)
-		newdart.damage = 5
-		newdart.nodamage = FALSE
 	newdart.damage_type = damage_type
 	if(pen)
 		newdart.pen = pen
 		pen.forceMove(newdart)
 		pen = null
+		newdart.damage = 5
+		newdart.nodamage = FALSE
```
Comment (ATH1909) [inline code/modules/projectiles/projectile/reusable/foam_dart.dm:30]:
```suggestion
		newdart.nodamage = FALSE
		newcasing.harmful = TRUE
```

Comment (SkeletalElite) [reply code/modules/projectiles/projectile/reusable/foam_dart.dm:30]: I presume this is for pacifism? But even so, It does not appear necessary, pacificists are unable to shoot recovered projectiles even without this addition in my testing

**Comment by Mothblocks** [comment]
Merge from upstream

**Pull request merged**